diff --git a/moveit2/Dockerfile b/moveit2/Dockerfile
index 301cf32..ccdd405 100644
--- a/moveit2/Dockerfile
+++ b/moveit2/Dockerfile
@@ -19,7 +19,7 @@
 #   VCS_REF     - The git revision of the Space ROS source code (no default value).
 #   VERSION     - The version of Space ROS (default: "preview")
 
-FROM osrf/space-ros:latest
+FROM osrf/space-ros:humble-2024.07.0
 
 # Define arguments used in the metadata definition
 ARG VCS_REF
@@ -91,13 +91,17 @@ RUN python3 -m pip install -U \
 
 # Get the MoveIt2 source code
 WORKDIR ${HOME_DIR}
-RUN sudo git clone https://github.com/ros-planning/moveit2.git -b ${ROSDISTRO} moveit2/src
+RUN sudo git clone https://github.com/ros-planning/moveit2.git ${MOVEIT2_DIR}/src \
+  && sudo chown -R ${USERNAME}:${USERNAME} ${MOVEIT2_DIR}/src \
+  && cd moveit2/src \
+  && git checkout 69c62f8114968c41857543d615a97eb36fdbaa97
+
 RUN cd ${MOVEIT2_DIR}/src \
-  && sudo git clone https://github.com/ros-planning/moveit2_tutorials.git -b ${ROSDISTRO}
+  && sudo git clone https://github.com/ros-planning/moveit2_tutorials.git \
+  && sudo chown -R ${USERNAME}:${USERNAME} moveit2_tutorials \
+  && cd moveit2_tutorials \
+  && git checkout 3e6576410e3322c86023ce9b35359fc73afc0de5
 
-# Update the ownership of the source files (had to use sudo above to work around
-# a possible inherited 'insteadof' from the host that forces use of ssh
-RUN sudo chown -R ${USERNAME}:${USERNAME} ${MOVEIT2_DIR}
 
 # Get rosinstall_generator
 RUN --mount=type=cache,target=/var/cache/apt,sharing=locked \
@@ -132,8 +136,6 @@ RUN --mount=type=cache,target=/var/cache/apt,sharing=locked \
  && rosdep install --from-paths ../spaceros/src src --ignore-src --rosdistro ${ROSDISTRO} -r -y --skip-keys "console_bridge generate_parameter_library fastcdr fastrtps rti-connext-dds-5.3.1 urdfdom_headers rmw_connextdds ros_testing rmw_connextdds rmw_fastrtps_cpp rmw_fastrtps_dynamic_cpp composition demo_nodes_py lifecycle rosidl_typesupport_fastrtps_cpp rosidl_typesupport_fastrtps_c ikos diagnostic_aggregator diagnostic_updater joy qt_gui rqt_gui rqt_gui_py"
 
 # Apply a patch to octomap_msgs to work around a build issue
-COPY --chown=${USERNAME}:${USERNAME} octomap_fix.diff ./src/octomap_msgs
-RUN cd src/octomap_msgs && git apply octomap_fix.diff
 
 # Build MoveIt2
 RUN /bin/bash -c 'source ${SPACEROS_DIR}/install/setup.bash \
diff --git a/space_robots/Dockerfile b/space_robots/Dockerfile
index 561802c..77049e2 100644
--- a/space_robots/Dockerfile
+++ b/space_robots/Dockerfile
@@ -48,7 +48,9 @@ ARG DEBIAN_FRONTEND=noninteractive
 # the cache won't make it into the built image but will be maintained between steps.
 RUN --mount=type=cache,target=/var/cache/apt,sharing=locked \
   --mount=type=cache,target=/var/lib/apt,sharing=locked \
-  sudo apt-get update -y && sudo apt-get install -y python3-rosinstall-generator
+  sudo apt-get update -y && sudo apt-get install -y python3-rosinstall-generator git-lfs
+
+RUN git lfs install
 
 # TODO(anyone): remove demo-pkgs.txt, no packages left after exclusions
 # Generate repos file for demo dependencies, excluding packages from Space ROS core.
diff --git a/space_robots/demo_manual_pkgs.repos b/space_robots/demo_manual_pkgs.repos
index 73b5481..5ea3e30 100644
--- a/space_robots/demo_manual_pkgs.repos
+++ b/space_robots/demo_manual_pkgs.repos
@@ -2,7 +2,7 @@ repositories:
   demos:
     type: git
     url: https://github.com/space-ros/demos.git
-    version: main
+    version: 981bd607e530cb1123166f964c2735ce54684ef0
   gz_ros2_control:
     type: git
     url: https://github.com/ros-controls/gz_ros2_control.git
@@ -26,7 +26,7 @@ repositories:
   simulation:
     type: git
     url: https://github.com/space-ros/simulation.git
-    version: main
+    version: ead69c448cd08484b12c252d17cbd8473c4d309e
   ros-humble-warehouse-ros-mongo:
     type: git
     url: https://github.com/ros-planning/warehouse_ros_mongo.git
